tests/test-work.lua

```diff
@@ -15,11 +15,13 @@ return require('lib/tap')(function (test)
             assert(n*n==r)
             assert(#s==4096)
             if step < count then
-                _uv.queue_work(ctx,n,s)
+                if ctx then _uv.queue_work(ctx,n,s) end
```

Comment:
Why is this change needed? Does this imply a breaking change in how `work`/`threadpool` is used?